sphinx_gallery/backreferences.py

```diff
@@ -155,8 +155,11 @@ def scan_used_functions(example_file, gallery_conf):
     example_code_obj = identify_names(example_file)
     if example_code_obj:
         codeobj_fname = example_file[:-3] + '_codeobj.pickle'
-        with open(codeobj_fname, 'wb') as fid:
-            pickle.dump(example_code_obj, fid, pickle.HIGHEST_PROTOCOL)
+        if (not os.path.isfile(codeobj_fname) or
+                os.stat(example_file).st_mtime >
+                os.stat(codeobj_fname).st_mtime):
```

Comment:
We abandoned checking against timestamps long ago. They lead to lots of problems. I Try using the change against the md5sum of the sourcefile.

Reply:
... actually walk me through this.  The source and destination files are not the same in the `.example` case.  Are you sugesting caching the `md5sum` of the source files between runs somewhere?

Reply:
@Titan-C I'm not sure how this could be done unless we embed the MD5s from the previous run somewhere. I guess we could start doing this.

It seems like we could at least allow a default `rebuild = True` default config arg (current behavior), or `rebuild='mtime'` (proposed changes here), and later we could add `rebuild='md5'` which would somehow keep track of the MD5 sums. That would provide a way forward without adding too many new complications for people who are satisfied with mtime.